src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -820,15 +823,16 @@ macro_rules! tool_extended {
                     compiler: $sel.compiler,
                     target: $sel.target,
                     tool: $tool_name,
-                    mode: if false $(|| $tool_std)? { Mode::ToolStd } else { Mode::ToolRustc },
+                    mode: $mode,
                     path: $path,
                     extra_features: $sel.extra_features,
                     source_type: SourceType::InTree,
                     allow_features: concat!($($allow_features)*),
                 });
 
-                if (false $(|| !$add_bins_to_sysroot.is_empty())?) && $sel.compiler.stage > 0 {
-                    let bindir = $builder.sysroot($sel.compiler).join("bin");
+                if (false $(|| !$add_bins_to_sysroot.is_empty())?) {
```

Comment:
Now that we put it into the correct stage, there's nothing wrong with `x build clippy --stage 0` anymore, and we can put that into the sysroot too.

Reply:
This conflicts with the idea of `ToolStd` and `ToolRustc` https://github.com/rust-lang/rust/blob/029cb1b13b6388b95e64e8996ec8b41a9f3cf16e/src/bootstrap/src/lib.rs#L257-L266

Reply:
I don't quite understand what you mean with this. The two `ToolStd` built here don't have any bins added to the sysroot anyways, so this doesn't affect them. As for the `ToolRustc`, this change was targetted specifically at them.